src/DeserializerFactory.php

```diff
@@ -29,45 +29,33 @@ class DeserializerFactory {
 	 */
 	private $legacyFactory;
 
-	/**
-	 * @var CurrentDeserializerFactory
-	 */
-	private $currentFactory;
-
 	public function __construct( Deserializer $dataValueDeserializer, EntityIdParser $idParser ) {
 		$this->legacyFactory = new LegacyDeserializerFactory( $dataValueDeserializer, $idParser );
-		$this->currentFactory = new CurrentDeserializerFactory( $dataValueDeserializer, $idParser );
 	}
 
 	/**
+	 * @param DispatchableDeserializer $currentEntityDeserializer
+	 *
 	 * @return Deserializer
 	 */
-	public function newEntityDeserializer() {
+	public function newEntityDeserializer( DispatchableDeserializer $currentEntityDeserializer ) {
```

Comment:
Maybe we should pass the deserializer in the constructor? That would perhaps make more sense..